design/remote-management.md

```diff
@@ -0,0 +1,66 @@
+# Remotely Managing a Cluster with the Package Controller
+
+The Package Controller should be able to run on the management cluster and install, update and delete packages on a workload cluster. The package custom resources for workload cluster will exist on the management cluster. The package custom resources for workload cluster will exist on the management cluster.
+
+# PackageBundleController (PBC) Custom Resource
+The management cluster will have a PackageBundleController custom resource for each workload cluster. The name of the PBC will be the name of the workload cluster.  This will allow each cluster to have different active bundles, different Kubernetes versions, and potentially different source registries. For example, the PBC for the billy cluster:
+
+```
+apiVersion: packages.eks.amazonaws.com/v1alpha1
+kind: PackageBundleController
+metadata:
+  name: billy
+  namespace: eksa-packages
+spec:
+  logLevel: 4
+  activeBundle: "v1-21-31"
+  upgradeCheckInterval: "24h"
+  source:
+    registry: public.ecr.aws/eks-anywhere
+    repository: eks-anywhere-packages-bundles
+```
+Currently, the PBC is in the eksa-packages namespace and must be named "eksa-packages-bundle-controller". This will have to be changed to support any cluster name in that namespace.
+
+# PackageBundle (bundle) Custom Resource
+We currently set the bundle state to show the user which bundle is active. Since we will have several clusters sharing the same set of bundles, we should remove this state and have the user look at the PBC to determine the active bundle. We may keep bundle state for invalid bundle names for instance.
+
+# Package Custom Resource
+A cluster specific namespace will need to be created for package custom resources in the form "eksa-packages-${cluster_name}". We use the name of the custom resource as the name of the installed package, so this will allow the same name across clusters. For example, to install the hello app on the billy cluster:
```

Comment:
What is the plan for this namespace when a workload cluster is removed? In that case the packages installed there will be uninstalled so we should get rid of the Package resource in this namespace in case the user tries to create a new workload cluster with the same name. Or we could just delete the namespace outright.

Reply:
When a workload cluster is deleted, the namespace should be deleted. I'll add a sentence about that below